File: DynamicTabsViewpager/app/src/main/java/com/kusu/dynamictabsviewpager/MainActivity.java
package com.kusu.dynamictabsviewpager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;

import android.os.Bundle;

import com.kusu.dynamictabsviewpager.databinding.ActivityMainBinding;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        List<Fragment> fragmentList = new ArrayList<>();
        int len = dummyTabs().size();
        for (int i = 0; i < len; i++) {
            ContentFragment contentFragment = new ContentFragment();
            Bundle bundle = new Bundle();
            bundle.putString("TAB", dummyTabs().get(i));
            contentFragment.setArguments(bundle);
            fragmentList.add(contentFragment);
        }



        MyFragmentPageAdapter myFragmentPageAdapter = new MyFragmentPageAdapter(this,
                getSupportFragmentManager(),
                fragmentList,
                dummyTabs());

        binding.viewpager.setAdapter(myFragmentPageAdapter);
        binding.tablayout.setupWithViewPager(binding.viewpager);
    }

    private ArrayList<String> dummyTabs() {
        ArrayList<String> tabs = new ArrayList<>();
        tabs.add("TabOne");
        tabs.add("TabTwo");
        tabs.add("TabThree");
        return tabs;
    }
}

File: Open_A_app_to_B_app/BTest/app/src/main/java/com/btest/MainActivity.kt
package com.btest

import android.database.DatabaseUtils
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.databinding.DataBindingUtil
import com.btest.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

            val data = intent?.getStringExtra("ATEST")
            binding.name.text = data
    }
}